mucommander-core-preload/src/main/java/com/mucommander/preload/PreloadedJFrame.java

```diff
@@ -0,0 +1,99 @@
+/*
+ * This file is part of muCommander, http://www.mucommander.com
+ *
+ * muCommander is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * muCommander is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package com.mucommander.preload;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.awt.LayoutManager;
+import java.util.Queue;
+import java.util.concurrent.ConcurrentLinkedDeque;
+
+import javax.swing.JFrame;
+import javax.swing.JPanel;
+
+/**
+ * A class that extends JFrame to be later on used by core.
+ * Since this bundle is loaded almost as the first (as it has no deps), it can be
+ * used by core having JFrame "preloaded" aka "cached" in JVM.
+*/
+public class PreloadedJFrame extends JFrame {
```

Comment:
I see that you've made quite a few changes to use this class but all it does is saving us from calling few constructors from the mucommander-core bundle? is it really significant?

Reply:
The main thing is that first initialization of any more complicated components of SWT/AWT is painfully slow (really slow), that's why I start building a couple of JFrames/JPanels as early as possible so when all the bundles are up, we just can put components in it, and SWT/AWT libs are also warmed-up already and therefore execute quickly as they should.

Preloading of those JFrames/JPanels (while OSGi/Felix is still trying to load all the staff) gave a lot of boost.

Reply:
ack
so why not to define it as a proper OSGi bundle with an activator class?

Reply:
because it takes around 1-2s to load OSGi framework (felix) and then to load jars bundles. While waiting for Felix to load we can use that time to init SWT/AWT in other thread (even my 10yr macbook has two cores, effectively 4 hyper-thread cores - we can employ them :)).